tests/testthat/test-generate_dist.R

```diff
@@ -70,14 +70,19 @@ test_that("'uniform' formula checked correctly", {
       x
     }),
     function(x) {
-      expect_silent(.genunif(x$n, x$range, NULL, environment()))
+      expect_silent(defData(varname = "z", formula = x$range, dist = "uniform"))
     }
   )
 
-  expect_warning(.genunif(3, "1;1", NULL, environment()), "are equal")
-  expect_error(.genunif(3, "", NULL, environment()), "format")
-  expect_error(.genunif(3, "1;2;3", NULL, environment()), "format")
-  expect_error(.genunif(3, "2;1", NULL, environment()), "'max' < 'min'")
+  expect_error(defData(varname = "z", formula = NULL, dist = "uniform", "format"))
+  expect_error(defData(varname = "z", formula = "1;2;3", dist = "uniform"), "format")
```

Comment:
The idea here was to test the different .gen* functions directly but your approach is actually correct as testing internal functions is discourage to prevent testing implementations instead of apis/behaviours. so :+1:

Reply:
Yeah - I realize that this was a fundamental issue, and I wasn't certain how to handle it. Once I started doing this, I noticed that this might require many more changes. And it was at this point, that I just stopped altogether, because I wasn't really sure about the way to proceed.